.travis.yml

```diff
@@ -9,22 +9,30 @@ addons:
       - libqt4-opengl-dev
       - g++-4.8
       - python3
+      - python-pip
+      - python3-pip
 compiler:
   - clang
 env:
   - py=python2
   - py=python3
-install: 
+install:
   - export NUMBER_OF_PROCESSORS=4
-  - export PATH=`pwd`/release/bin:`pwd`/scripts:${PATH}
+  - export PATH=`pwd`/bin:${PATH}
   - export EIGEN_CFLAGS=-I`pwd`/../eigen
   - (cd ..; hg clone https://bitbucket.org/eigen/eigen/; cd eigen; hg update 3.3)
+  - pip install recommonmark sphinx sphinx-rtd-theme
+  - pip3 install recommonmark sphinx sphinx-rtd-theme
 script:
+  - ${py} -m sphinx -n -N -W -w sphinx.log -c docs/ docs/ tmp/
   - ./check_memalign && $py ./configure -assert && $py ./build -nowarnings && ./run_tests
+  - ./docs/generate_user_docs.sh && ! ( git diff -- . ':(exclude)docs/reference/commands/mrview.rst' ':(exclude)docs/reference/commands/shview.rst' | tee gitdiff.log | grep '^-\|^+' | grep -q -v 'commands_list.rst\|mrview\|shview' )
```

Comment:
By the way, I reckon it would be cleaner to use the `--stat` or `--numstat` option to `git diff` to check for differences - much cleaner to parse (just need to check for non-empty output)... Something like:
```
./docs/generate_user_docs.sh && ( 
  gitdiff=$(git diff --numstat -- docs ':!docs/reference/commands/??view.rst' ); 
  [[ $gitdiff ]] && ( 
    git diff > gitdiff.log; 
    echo "documentation is not up to date - please run docs/generate_user_docs.sh and commit again"; 
    echo $gitdiff 
  ) 
)

```

Does this achieve the same thing...?